ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala

```diff
@@ -195,9 +196,51 @@ private[index] class IndexServiceImpl(
       .buffered(metrics.daml.index.completionsBufferSize, LedgerApiStreamsBufferSize)
 
   override def getActiveContracts(
-      filter: TransactionFilter,
+      f: TransactionFilter,
       verbose: Boolean,
-  )(implicit loggingContext: LoggingContext): Source[GetActiveContractsResponse, NotUsed] =
+  )(implicit loggingContext: LoggingContext): Source[GetActiveContractsResponse, NotUsed] = {
+    val originalVerbose = TimerX.Original.verboseEnriching.reset()._1
+    val originalDBDeserialization = TimerX.Original.dbDeserialization.reset()._1
+    val originalApiSerialization = TimerX.Original.apiSerialization.reset()._1
+    val originalTotal = originalApiSerialization + originalDBDeserialization + originalVerbose
+    val computeInterfaceView = TimerX.InterfaceProjection.computeInterfaceView.reset()._1
+    val serializeInterfaceView = TimerX.InterfaceProjection.apiSerialization.reset()._1
+    val interfaceTotal = computeInterfaceView + serializeInterfaceView
+    println(s"""
+         |Original contract_argument computation: $originalTotal ms
+         |  verbose:              $originalVerbose ms
+         |  db-deserialization:   $originalDBDeserialization ms
+         |  api-serialization:    $originalApiSerialization ms
+         |Interface projection computation:       $interfaceTotal ms
+         |  compute view:         $computeInterfaceView ms
+         |  serialize view:       $serializeInterfaceView ms
+         |
+         |""".stripMargin)
+
+    val filter =
+      if (false) f
+      else
+        TransactionFilter(
```

Comment:
Just to confirm my understanding: You're using this so you don’t have to actually make the benchtool send requests for interface subscriptions and instead you can just send whatever and it turns into an interface subscriptions request?

Reply:
yupp, this is the easiest thing I could figure out compared to build full fledged interface subscription support for bench-tool